src/main/java/com/example/gymApp/service/TraineeService.java

```diff
@@ -99,6 +107,44 @@ public Trainee updateTrainee(Trainee trainee, String newName, String newLastName
 
 
   }
+
+  @Transactional
+  public void deleteTraineeByUsername(String username) {
+
+    Optional<User> userOpt = userRepository.findByUsername(username);
+
+    if (userOpt.isEmpty()) {
+      throw new NoSuchElementException("No user found with the provided username");
+    }
+
+    User user = userOpt.get();
+
+    Optional<Trainee> traineeOpt = traineeRepository.findByUser(user);
+
+    if (traineeOpt.isEmpty()) {
+      throw new NoSuchElementException("No trainee found for the provided user");
+    }
+    Trainee trainee = traineeOpt.get();
```

Comment:
You can use `orElseThrow`
`Trainee trainee =  traineeRepository.findByUser(user). orElseThrow(()->new NoSuchElementException("No trainee found for the provided user"))`